molecule/model/schema_v2.py

```diff
@@ -944,6 +945,20 @@ class Validator(cerberus.Validator):
     def __init__(self, *args, **kwargs):
         super(Validator, self).__init__(*args, **kwargs)
 
+    def _validate_unique(self, unique, field, value):
+        """Ensure value uniqueness.
+
+        The rule's arguments are validated against this schema:
+        {'type': 'boolean'}
+        """
+        if unique:
+            root_key = self.schema_path[0]
+            data = [doc[field] for doc in self.root_document[root_key]]
```

Comment:
Since you don't process data twice, use a generator expression which will allow lazy iteration without the creation of an intermediate list.
```suggestion
            data = (doc[field] for doc in self.root_document[root_key])
```